docs/src/pages/layout/use-media-query/use-media-query.md

```diff
@@ -0,0 +1,119 @@
+## Using Material-UI's breakpoint helpers
+
+You can use the Material-UI's [breakpoint helpers](/layout/breakpoints/) as follow:
+
+```jsx
+import { useTheme } from '@material-ui/styles';
+import { unstable_useMediaQuery as useMediaQuery } from '@material-ui/core/useMediaQuery';
+
+function MyComponent() {
+  const theme = useTheme();
+  const matches = useMediaQuery(theme.breakpoints.up('sm'));
+
+  return <span>{`theme.breakpoints.up('sm') matches: ${matches}`}</span>;
+}
+```
+
+{{"demo": "pages/layout/use-media-query/ThemeHelper.js", "react": "next"}}
+
+## Server side rendering
+
+You need to use a [matchMedia](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) ponyfill on the server, we recommand using [css-mediaquery](https://github.com/ericf/css-mediaquery).
+We also encourage the usage of the `unstable_useMediaQueryTheme` version of the hook that fetches properties from the theme. This way, you can provide a `ssrMatchMedia` option once for all your React tree.
+
+{{"demo": "pages/layout/use-media-query/ServerSide.js", "react": "next"}}
+
+## Migrating from `withWidth()`
+
+The `withWidth()` higher-order component inject the screen width of the page.
+You can reproduce the same behavior as follow:
+
+```jsx
+function MyComponent() {
+  const theme = useTheme();
+  const width =
+    [...theme.breakpoints.keys].reverse().reduce((output, key) => {
+      const matches = useMediaQuery(theme.breakpoints.only(key));
+
+      return !output && matches ? key : output;
+    }, null) || 'xs';
+
+  return <span>{width}</span>;
+}
+```
+
+{{"demo": "pages/layout/use-media-query/UseWidth.js", "react": "next"}}
+
+## API
+
+### `unstable_useMediaQuery(query, [options]) => matches`
+
+#### Arguments
+
+1. `query` (*String*): A string representing the media query to handle.
+2. `options` (*Object* [optional]):
+  - `options.defaultMatches` (*Boolean* [optional]):
+  As `window.matchMedia()` is unavailable on the server,
+  we return a default matches during the first mount. The default value is `false`.
+  - `options.noSSR` (*Boolean* [optional]): Defaults to `false`.
+  In order to perform the server-side rendering reconciliation, it needs to render twice.
+  A first time with nothing and a second time with the children.
+  This double pass rendering cycle comes with a drawback. It's slower.
+  You can set this flag to `true` if you are **not doing server-side rendering**.
+  - `options.ssrMatchMedia`
+  In some situation, you might want to use an heuristic to approximate
+  the screen of the client browser.
+  For instance, you could be using the user-agent or the client-hint https://caniuse.com/#search=client%20hint.
+  You can provide a global ponyfill using [`custom properties`](/customization/themes/#properties) on the theme. Check the [server-side rendering example](#server-side-rendering).
+
+#### Returns
+
+`matches`: The matches is a Boolean that returns true if the document currently matches the media query, or false if not.
```

Comment:
Matches is `true` if the document currently matches the media query and `false` when it does not.